src/lib.rs

```diff
@@ -94,39 +94,63 @@ impl fmt::Display for ActorError {
 
 impl std::error::Error for ActorError {}
 
-/// Reasons why sending a message to an actor can fail.
-#[derive(Debug)]
-pub enum SendError {
-    /// The channel's capacity is full.
-    Full,
-    /// The recipient of the message no longer exists.
-    Disconnected,
+/// Failures that can occur when sending a message to an actor.
+#[derive(Debug, Clone, Copy)]
+pub struct SendError {
+    recipient_name: &'static str,
+    reason: SendErrorReason,
+}
+
+impl SendError {
+    /// Get the name of the intended recipient.
+    pub fn recipient_name(&self) -> &'static str {
```

Comment:
I remember having this conversation in some other PR, and we said we usually prefer making the fields `pub` instead of adding getters unless changing a value breaks an invariant?

Reply:
The conversation was probably this one: https://github.com/tonarino/actor/pull/53#discussion_r634371138 (+ related #56).

I still don't have any clear opinion on this. Likely this case would fit to "compound, passive data structure" per [Rust API guidelines](https://rust-lang.github.io/api-guidelines/future-proofing.html#c-struct-private)?
> Public fields are most appropriate for struct types in the C spirit: compound, passive data structures. Otherwise, consider providing getter/setter methods and hiding fields instead.

This type being an _error_ type, I think I was subconsciously affected by https://matklad.github.io/2020/10/15/study-of-std-io-error.html and its praise of the `std::io::Error`.

Thinking about it more, I think I slightly lean towards making the fields public. The API stability much smaller concern for us than `std`.

Reply:
Yeah, `io::Error` type comes with accessors, and I was also wondering whether it's where you sought for consistency. I don't feel strongly either way, but I wanted to ask your opinion, so I can start forming my habits.

Reply:
I ended up making the fields `pub`.

Rationale: if we ever want to contain original message `M` as a payload in the error, it would be more handy if it is pub so that users can move the message out of the error (without introducing special methods for it).